packages/darklang/languageTools/writtenTypesToProgramTypes.dark

```diff
@@ -16,60 +16,42 @@ module Darklang =
-            | "PACKAGE" :: owner :: rest ->
-              (ProgramTypes.FQName.Package
-                { owner = owner
-                  modules = rest
-                  name = ProgramTypes.TypeName.Name.TypeName i.typ.name
-                  version = 0L })
-              |> ProgramTypes.FQName.FQName.Package
-
-            // | "Builtin" :: rest ->
-            //   (ProgramTypes.FQName.BuiltIn
-            //     { modules = rest
-            //       name = ProgramTypes.TypeName.Name.TypeName i.typ.name
-            //       version = 0L })
-            //   |> ProgramTypes.FQName.FQName.BuiltIn
-
-            | modules ->
-              (ProgramTypes.FQName.UserProgram
-                { modules = modules
-                  name = ProgramTypes.TypeName.Name.TypeName i.typ.name
-                  version = 0L })
-              |> ProgramTypes.FQName.FQName.UserProgram
+            let currentModule = Stdlib.List.map i.modules (fun (m, _) -> m.name)
+            let nametoResolve = [ i.typ.name ]
+
+            let resolvedName =
+              NameResolver.TypeName.resolve
+                resolver
+                currentModule
+                (WrittenTypes.Name.Unresolved nametoResolve)
+
+            match resolvedName with
+            | Ok n -> n
+            | Error e -> e
 
         module Fn =
           let toPT (fn: WrittenTypes.FnIdentifier) : String = fn.name
 
         module QualifiedFn =
           let toPT
+            (resolver: NameResolver.NameResolver)
             (i: WrittenTypes.QualifiedFnIdentifier)
             : ProgramTypes.FQName.FQName =
-            match Stdlib.List.map i.modules (fun (m, _) -> m.name) with
-            | "PACKAGE" :: owner :: rest ->
-              (ProgramTypes.FQName.Package
-                { owner = owner
-                  modules = rest
-                  name = ProgramTypes.FnName.Name.FnName i.fn.name
-                  version = 0L })
-              |> ProgramTypes.FQName.FQName.Package
-
-            | "Builtin" :: rest ->
-              (ProgramTypes.FQName.BuiltIn
-                { modules = rest
-                  name = ProgramTypes.FnName.Name.FnName i.fn.name
-                  version = 0L })
-              |> ProgramTypes.FQName.FQName.BuiltIn
-
-            | modules ->
-              (ProgramTypes.FQName.UserProgram
-                { modules = modules
-                  name = ProgramTypes.FnName.Name.FnName i.fn.name
-                  version = 0L })
-              |> ProgramTypes.FQName.FQName.UserProgram
+            let modules = Stdlib.List.map i.modules (fun (m, _) -> m.name)
+            let nametoResolve = [ i.fn.name ]
+
+            let resolvedName =
+              NameResolver.FnName.resolve
+                resolver
+                modules
+                (WrittenTypes.Name.Unresolved nametoResolve)
+
+            match resolvedName with
+            | Ok name -> name
+            | Error e -> e
```

Comment:
Hmm this seems a little bit suspicious. Probably OK to keep, but feels like some TODO would be appropriate here